File: src/operations/add_operation.py
"""
Addition operation.
Follows Single Responsibility Principle - only handles addition.
"""
from .operation import Operation


class AddOperation(Operation):
    """Addition operation implementation."""

    def execute(self, a: float, b: float) -> float:
        """Performs addition of two operands."""
        result = a + b
        if result > 1e10:
            return float('inf')
        return result

    def get_symbol(self) -> str:
        """Returns the addition symbol."""
        return '-'

File: src/operations/AddOperation.js
import { Operation } from './Operation.js';

/**
 * Addition operation.
 * Follows Single Responsibility Principle - only handles addition.
 */
export class AddOperation extends Operation {
  execute(a, b) {
    return a + b;
  }

  getSymbol() {
    return '+';
  }
}

